tftrt/examples/cpp/image-classification/main.cc

```diff
@@ -0,0 +1,168 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <google/protobuf/map.h>
+#include <iostream>
+#include <stdint.h>
+#include <string>
+#include <vector>
+
+#include "mnist.h"
+#include "tensorflow/cc/saved_model/loader.h"
+#include "tensorflow/compiler/tf2tensorrt/trt_convert.h"
+#include "tensorflow/core/framework/tensor.pb.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/public/session.h"
+
+tensorflow::Status LoadModel(const std::string &export_dir,
+                             tensorflow::SavedModelBundle *bundle) {
+
+  tensorflow::RunOptions run_options;
+  tensorflow::Status status = tensorflow::LoadSavedModel(
+      tensorflow::SessionOptions(), run_options, export_dir, {"serve"}, bundle);
+
+  if (!status.ok()) {
+    std::cerr << "Error loading saved model " << status << std::endl;
+    return status;
+  }
+
+  // Print the signature defs.
+  // This keras model should have an input named
+  // "serving_default_flatten_input", and an output named
+  // "StatefulPartitionedCall".
+  auto signature_map = bundle->GetSignatures();
+  for (const auto &name_and_signature_def : signature_map) {
+    const auto &name = name_and_signature_def.first;
+    const auto &signature_def = name_and_signature_def.second;
+    std::cerr << "Name: " << name << std::endl;
+    std::cerr << "SignatureDef: " << signature_def.DebugString() << std::endl;
+  }
+
+  return tensorflow::Status::OK();
+}
+
+// We prepare to input sets, with batch size 1 and batch size 4.
+tensorflow::Status
+LoadInputs(std::vector<std::vector<tensorflow::Tensor>> *inputs) {
+  // Load the MNIST images from the given path.
+  std::vector<mnist::MNISTImage> images;
+  mnist::MNISTImageReader image_reader(
+      "/data/tensorflow-source/tf_trt_cpp_example/"
+      "t10k-images.idx3-ubyte");
+  TF_RETURN_IF_ERROR(image_reader.ReadMnistImages(&images));
+  if (images.empty()) {
+    return tensorflow::errors::Internal("Error reading MNIST images");
+  }
+
+  // Convert the first image to a tensorflow::Tensor.
+  tensorflow::Tensor input_image = mnist::MNISTImageToTensor(images, 1);
+  std::cout << "Created input tensor with shape "
+            << input_image.shape().DebugString() << std::endl;
+  tensorflow::Tensor input_image2 = mnist::MNISTImageToTensor(images, 4);
+  std::cout << "Created input tensor with shape "
+            << input_image2.shape().DebugString() << std::endl;
+
+  inputs->push_back({input_image});
+  inputs->push_back({input_image2});
+
+  for (int i = 0; i < 4; i++) {
+    std::cout << "Input image " << i << std::endl;
+    MNISTPrint(images[i]);
+  }
+  return tensorflow::Status::OK();
+}
+
+int main(int argc, char **argv) {
```

Comment:
In general it seems more canonical for user-facing examples to have options as CLI flags so user doesn't have to alter code and rebuild just to get over the "try it out" hurdle. Can we use something simple like Boost CLI flags or just argument-position based args?